### PR TITLE
access to raw update json added

### PR DESCRIPTION
**Pull request by Narrator69**
very useful thing

### CODE REVIEW

**Comment by zhuharev** [comment]
Thank you for your contribution. You are a little behind the master, we use `json.Decoder` if the debug is off and decode json without buffer allocation:
https://github.com/go-telegram-bot-api/telegram-bot-api/blob/6b5102fab5586322d6fe98112fc671ad23dfae33/bot.go#L98-L99

Also I do not know the application of your field, because the update has a certain format.

**Comment by zhuharev** [comment]
Oh, I did not see that you made a correction only in webhook, this will create ambiguous behavior `GetUpdatesChan` and `ListenForWebhook`

File: types.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ type Update struct {
 	CallbackQuery      *CallbackQuery      `json:"callback_query"`
 	ShippingQuery      *ShippingQuery      `json:"shipping_query"`
 	PreCheckoutQuery   *PreCheckoutQuery   `json:"pre_checkout_query"`
+	RawUpdateJson      string
```
Comment (zhuharev) [inline types.go]: Use capital letters for abbreviation JSON

**Comment by Narrator69** [comment]
Hey! Thanks for feedback. I made the changes you asked for. This feature is useful because allows to store raw data in db and analyze/debug it later (as example)

**Comment by zhuharev** [comment]
I think we need to discuss, because yours `update.RawUpdateJSON` equals `json.Marshal(update)`. I doubt that many people need a query entirely, and this affects the speed and allocation of memory.

**Comment by Narrator69** [comment]
What do you think about variable like `collectRawUpdates (bool)`?

File: bot.go [hunk 2]
```diff
@@ -423,6 +423,14 @@ func (bot *BotAPI) GetUpdates(config UpdateConfig) ([]Update, error) {
 	var updates []Update
 	json.Unmarshal(resp.Result, &updates)
 
+	// Every update contains raw JSON which
+	// can include information about other
+	// updates. This is because response is
+	// single for all. Be aware.
+	for _, v := range updates {
+		v.RawUpdateJSON = string(resp.Result)
```
Comment (Syfaro) [inline bot.go]: `resp.Result` is a byte array (although typed to `json.RawMessage`), instead of converting it to a string you could use it directly.

**Pull request closed without merging**